### ISSUE TITLE
Letterhead support

### ISSUE THREAD

**Issue by elmcrest** [bpampuch/pdfmake#1377]
Hey Bartek,

I've just stumbled upond pdfmake and it feels *really* promising to me just from using the playground. So, let me first say congratulation for what you already have, maybe including some credits to PDFKit but however, one special thing which breaks f.e. python's reportlab is a single table cell which breaks overrr pages, but pdfmake just renders and inserts a pagebreak, really cool.

however, for my use case, generating proposals and invoices and such stuff, I'd really need the ability to place a background letterhead PDF into the resulting PDF. Ideally it would also decide to use the first page of a letterheadPDF for the first page of the targetPDF and the second page of the letterheadPDF forr the remaining targetPDF pages.

Is this already supported?
If not, what would you think, how this could be achieved? I'm really not a JS/TS dev but maybe I could help somehow...

(PS: couldn't find anything related yet in the open and closed issues.)

**Comment by liborm85** [comment]
To create letterhead you can use [background layer](https://github.com/bpampuch/pdfmake#background-layer) and [headers and footers](https://github.com/bpampuch/pdfmake#headers-and-footers).

**Comment by elmcrest** [comment]
Ok cool I‘ve missed that so far. I‘m going to give it a try asap and report back here.

**Comment by elmcrest** [comment]
so, I've sucessfully tested it with a JPEG image but couldn't find a way to get a background pdf file working. can you give me a hint on this?
I've tried data:application/pdf;base64, data:image/pdf;base64, and some other variants...

**Comment by liborm85** [comment]
JPEG image have this definition:
> **data:image/jpeg;base64,**/9j/4RC.....

**Comment by Wabbala** [comment]
Inserting a pdf is not supported, try and convert it to JPEG/PNG first.

**Comment by elmcrest** [comment]
Yeah ok but then I loose the vector based nature und fonts, so it‘s not ideal, especially when the pdf is viewed an potentially zoomed on touch devices. 
Are there any plans to add support?

**Comment by liborm85** [comment]
PDF inserting, merging, etc. is not supported. No any plans to add support, because it is too problematic - pdfmake can't open pdf files, only write via pdfkit library.

**Comment by elmcrest** [comment]
Not really, no. At first sight, sufe, 300 DPI is enough to be printed, btw no matter the pagesize since it‘s a ‚per inch‘ unit. 
But, people are going to zoom in on mobile and tablets, and even at Desktops.
Anyways, ok, so the question about futute plans is still open, or is this a „no“?

**Comment by liborm85** [comment]
No. For loading PDF documents a different library would be needed, because pdfmake is only for writing pdf documents.

If you have a pdf document that contains images, vectors, etc. - create this elements manually in pdfmake docDefinition.
